### PR TITLE
Components Page - Internal Navigation

### PR DESCRIPTION
**Pull request by franckgaudin**
---------------------
Alex's edit:
links to test: 
https://deploy-preview-274--wl-hopper.netlify.app/components/getting-started/installation
https://deploy-preview-274--wl-hopper.netlify.app/components/actions/button

### CODE REVIEW

File: apps/docs/app/components/[...slug]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+export async function generateStaticParams(): Promise<PageProps["params"][]> {
+    const componentsDetails = await getComponentDetails();
+
+    return componentsDetails.map(({ slug }) => {
+        const [section, type] = slug.split("/");
+
+        return ({
+            slug: [section, type]
+        });
+    }
```
Comment (fraincs) [inline apps/docs/app/components/[...slug]/page.tsx:25]: Indentation seems off here

File: apps/docs/context/sidebar/SidebarProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import { createContext, useCallback, useContext, useEffect, useState } from "react";
+
+interface SidebarProps {
+    isSidebarOpen: boolean;
+    toggleSidebar: () => void;
+}
+
+const SidebarContext = createContext<SidebarProps>({
+    isSidebarOpen: false,
+    toggleSidebar: () => {
```
Comment (fraincs) [inline apps/docs/context/sidebar/SidebarProvider.tsx:13]: `toggleSidebar: () => {}`

**Comment by alexasselin008** [comment]
image

The heading changed in this PR,

**Comment by alexasselin008** [comment]
The theme toggle goes over the nav bar
image

**Review by alexasselin008 — CHANGES_REQUESTED** [review]
2 qa issues

**Review by alexasselin008 — APPROVED** [review]

**Comment by fraincs** [comment]
image
What does the * refers to?

**Comment by fraincs** [comment]
image
Clicking report an issue could pre fill the title with the component name.

**Pull request merged**